File: JavaScript_workspace/day13/Hero.js
class Hero extends GameObject{
    constructor(container, src, x, y, width, height, velX, velY){
        super(container, src, x, y, width, height, velX, velY); // 상속관계에 있는 자식의 생성자가 호출될 때에는 반드시 부모의 초기화 선행
    }
}

File: JAVA_workspace/project0304/src/com/sds/project0304/Hero.java
package com.sds.project0304;

// 게임의 최상위 객체인 GameObject를 상속받아
// 부모 메서드의 move() 메서드를 오버라이딩
public class Hero extends GameObject{
	// 추상메서드를 상속받는 자식은 부모가 진 빚을 물려받는 느낌으로 개발
	// 부모가 완성하지 못하였던 해당 메서드를 자식세대에서 청산할 의무를 가짐
	public void move() {
		// 움직임 코드 작성
	}
	
	public static void main(String[] args) {
		Hero h = new Hero();
	}
}
